src/h5web/visualizations/heatmap/hooks.ts

```diff
@@ -1,19 +1,101 @@
-import React, { useCallback, useRef, useEffect, CSSProperties } from 'react';
+import React, {
+  useCallback,
+  useRef,
+  useEffect,
+  CSSProperties,
+  useContext,
+  useMemo,
+} from 'react';
 import { Vector3 } from 'three';
 import { ReactThreeFiber, PointerEvent, useThree } from 'react-three-fiber';
 import { clamp } from 'lodash-es';
 import { useMeasure } from 'react-use';
-import { useHeatmapStore } from './store';
+import { extent } from 'd3-array';
+import { scaleSymlog, scaleLinear, scaleSequential } from 'd3-scale';
+import { rgb } from 'd3-color';
+import { useHeatmapConfig } from './config';
+import { HeatmapProps, HeatmapContext } from './HeatmapProvider';
+import { Domain, DataScale, D3Interpolator } from './models';
+import { INTERPOLATORS } from './interpolators';
 
 const ZOOM_FACTOR = 0.95;
 
-export function useHeatmapStyles<T>(
-  dims: [number, number],
-  axisOffset: [number, number]
-): [(elem: T) => void, CSSProperties | undefined] {
-  const [leftAxisWidth, bottomAxisHeight] = axisOffset;
+export function useProps(): HeatmapProps {
+  const props = useContext(HeatmapContext);
 
-  const keepAspectRatio = useHeatmapStore(state => state.keepAspectRatio);
+  if (!props) {
+    throw new Error('Missing Heatmap provider.');
+  }
+
+  return props;
+}
```

Comment:
This file is where things actually happen. Here, is the `useProps` hook that provides a convenient way to access the props (`dims`, `data`, `axisOffsets`) from the context.